katharsis-jpa/src/main/java/io/katharsis/jpa/internal/meta/impl/MetaCollectionTypeImpl.java

```diff
@@ -0,0 +1,41 @@
+package io.katharsis.jpa.internal.meta.impl;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.katharsis.jpa.internal.meta.MetaCollectionType;
+import io.katharsis.jpa.internal.meta.MetaType;
+
+public abstract class MetaCollectionTypeImpl extends MetaTypeImpl implements MetaCollectionType {
+
+	private MetaType elementType;
+
+	public MetaCollectionTypeImpl(MetaElementImpl parent, Class<?> implClass, Type implType, MetaType elementType) {
+		super(parent, implClass, implType);
+		this.elementType = elementType;
+	}
+
+	@Override
+	public MetaType getElementType() {
+		return elementType;
+	}
+
+	@Override
+	public Object fromString(String values) {
+		throw new UnsupportedOperationException("no yet implemented");
+	}
+
+	@Override
+	public <T> Collection<T> newInstance() {
+		if (getImplementationClass() == Set.class)
+			return new HashSet<T>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)